File: backend/app/schemas/release_center.py
from datetime import datetime
from pydantic import BaseModel, Field


class DiffItem(BaseModel):
    module: str
    path: str
    before: str | int | float | bool | None = None
    after: str | int | float | bool | None = None
    risk_level: str = "low"


class DiffResponse(BaseModel):
    from_revision: int
    to_draft: str
    changes: list[DiffItem] = Field(default_factory=list)


class PublishRequest(BaseModel):
    draft_id: str
    confirm_token: str  # 凭证痕迹 — 移除默认值"publish-confirmed"，必须由调用方显式传入
    publish_note: str | None = None


class PublishResponse(BaseModel):
    publish_id: str
    revision: int
    status: str
    published_at: datetime


class RollbackRequest(BaseModel):
    target_revision: int
    reason: str | None = None


# 补充 rollback 的 response_model，使接口返回类型明确
class RollbackResponse(BaseModel):
    status: str
    target_revision: int

File: frontend/src/api/releaseCenter.ts
import api from '@/utils/http'

export type DiffItem = {
  module: string
  path: string
  before: string | number | boolean | null
  after: string | number | boolean | null
  risk_level: string
}

export type DiffResponse = {
  from_revision: number
  to_draft: string
  changes: DiffItem[]
}

export type PublishResponse = {
  publish_id: string
  revision: number
  status: string
  published_at: string
}

export const getDraftDiff = async (draftId: string): Promise<DiffResponse> => {
  const res = await api.get(`/api/v1/release-center/drafts/${draftId}/diff`)
  return res.data
}

export const publishDraft = async (draftId: string, confirmToken: string, publishNote?: string): Promise<PublishResponse> => {
  const res = await api.post('/api/v1/release-center/publish', {
    draft_id: draftId,
    confirm_token: confirmToken,
    publish_note: publishNote || null
  })
  return res.data
}

// FIXED: rollback 返回类型与后端 RollbackResponse 对齐
export type RollbackResponse = {
  status: string
  target_revision: number
}

export const rollbackRevision = async (targetRevision: number, reason?: string): Promise<RollbackResponse> => {
  const res = await api.post('/api/v1/release-center/rollback', {
    target_revision: targetRevision,
    reason: reason || null
  })
  return res.data
}
